From e7da9f9a1d16420a9aca60bca5188d2d4c904cf0 Mon Sep 17 00:00:00 2001
From: "Chen, Yu" <yu.y.chen@intel.com>
Date: Mon, 24 Jan 2022 15:30:50 +0800
Subject: [PATCH] Don't use .symlink.timestamp as target

It will cause CTS JniStaticTest failed it this file exists in vendor/lib folders.

Signed-off-by: Chen, Yu <yu.y.chen@intel.com>
Change-Id: I78a6234b035688d9db525a47bba9335d059778a7
---
 android/mesa3d_cross.mk | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/android/mesa3d_cross.mk b/android/mesa3d_cross.mk
index 3d4e77e64b0..fd50d4c9a4e 100644
--- a/android/mesa3d_cross.mk
+++ b/android/mesa3d_cross.mk
@@ -299,13 +299,12 @@ endef
 
 $(foreach driver,$(BOARD_MESA3D_VULKAN_DRIVERS), $(eval $(call vulkan_target,$(driver))))
 
-$($(M_TARGET_PREFIX)TARGET_OUT_VENDOR_SHARED_LIBRARIES)/dri/.symlinks.timestamp: MESA3D_GALLIUM_DRI_DIR:=$(MESA3D_GALLIUM_DRI_DIR)
-$($(M_TARGET_PREFIX)TARGET_OUT_VENDOR_SHARED_LIBRARIES)/dri/.symlinks.timestamp: $(MESON_OUT_DIR)/install/.install.timestamp
+$($(M_TARGET_PREFIX)TARGET_OUT_VENDOR_SHARED_LIBRARIES)/dri/iris_dri.so: MESA3D_GALLIUM_DRI_DIR:=$(MESA3D_GALLIUM_DRI_DIR)
+$($(M_TARGET_PREFIX)TARGET_OUT_VENDOR_SHARED_LIBRARIES)/dri/iris_dri.so: $(MESON_OUT_DIR)/install/.install.timestamp
 	# Create Symlinks
 	mkdir -p $(dir $@)
 	ls -1 $(MESA3D_GALLIUM_DRI_DIR)/ | PATH=/usr/bin:$$PATH xargs -I{} ln -s -f libgallium_dri.so $(dir $@)/{}
-	touch $@
 
-$($(M_TARGET_PREFIX)MESA3D_GALLIUM_DRI_BIN): $(TARGET_OUT_VENDOR)/$(MESA3D_LIB_DIR)/dri/.symlinks.timestamp
+$($(M_TARGET_PREFIX)MESA3D_GALLIUM_DRI_BIN): $(TARGET_OUT_VENDOR)/$(MESA3D_LIB_DIR)/dri/iris_dri.so
 	echo "Build $@"
 	touch $@
-- 
2.33.1

